test/e2e/tests/test_pipeline_execution.py

```diff
@@ -0,0 +1,193 @@
+        expected_status: str,
+        wait_periods: int = 30,
+        period_length: int = 30,
+    ):
+        return wait_for_status(
+            expected_status,
+            wait_periods,
+            period_length,
+            get_pipeline_execution_resource_status,
+            reference,
+        )
+
+    def _wait_sagemaker_pipeline_execution_status(
+        self,
+        pipeline_execution_arn,
+        expected_status: str,
+        wait_periods: int = 30,
+        period_length: int = 30,
+    ):
+        return wait_for_status(
+            expected_status,
+            wait_periods,
+            period_length,
+            get_sagemaker_pipeline_execution_status,
+            pipeline_execution_arn,
+        )
+
+    def _assert_pipeline_execution_status_in_sync(
+        self, pipeline_execution_arn, reference, expected_status
+    ):
+        assert (
+            self._wait_sagemaker_pipeline_execution_status(
+                pipeline_execution_arn, expected_status
+            )
+            == self._wait_resource_pipeline_execution_status(reference, expected_status)
+            == expected_status
+        )
+
+    def test_pipeline_execution_succeeded(self, pipeline_execution):
+        (reference, spec, resource) = pipeline_execution
+        assert k8s.get_resource_exists(reference)
+
+        pipeline_name = resource["spec"].get("pipelineName")
+        # Need PipelineExecutionArn to reference the resource
+        pipeline_execution_arn = sagemaker_client().list_pipeline_executions(
+            PipelineName=pipeline_name
+        )["PipelineExecutionSummaries"][0]["PipelineExecutionArn"]
+
+        pipeline_execution_desc = get_sagemaker_pipeline_execution(
+            pipeline_execution_arn
+        )
+        if k8s.get_resource_arn(resource) is None:
+            logging.error(
+                f"ARN for this resource is None, resource status is: {resource['status']}"
+            )
+
+        assert k8s.get_resource_arn(resource) == pipeline_execution_arn
+
+        self._assert_pipeline_execution_status_in_sync(
+            pipeline_execution_arn, reference, cfg.JOB_STATUS_EXECUTING
+        )
+        assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "False")
+
+        # Update the resource
+        new_pipeline_execution_display_name = random_suffix_name(
+            "updated-display-name", 38
+        )
+        spec["spec"][
+            "pipelineExecutionDisplayName"
+        ] = new_pipeline_execution_display_name
+        resource = k8s.patch_custom_resource(reference, spec)
+        resource = k8s.wait_resource_consumed_by_controller(reference)
+        assert resource is not None
+
+        self._assert_pipeline_execution_status_in_sync(
+            pipeline_execution_arn, reference, cfg.JOB_STATUS_EXECUTING
+        )
+        assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "False")
+
+        pipeline_execution_desc = get_sagemaker_pipeline_execution(
```

Comment:
verify last modified time got updated

Reply:
sure with last-modified-time, afaik tags do not exist for this resource